### DISCUSSION TITLE
About Permissions Plugin

### DISCUSSION THREAD

**Discussion by Mr-Ao-Dragon** [fnuecke/oc2#223] [Ideas]
The first version of OpenCompuers' bots and drones did not work properly with plugins such as LuckPerms and did not recognize permissions correctly, I commissioned a compatibility plugin to solve the problem, but this is only a stop-gap measure and I hope to add compatibility with the permission system.
As an example: the robot's working mechanism relies on Fake-Player, which I cannot operate on the bukkit level, such as giving it inheritance rights, so the player assembles, and the placed robot cannot do any interaction where PlotSquard is running.
I'd like to add a compatibility layer directly and register permission nodes so that not only can the bots work properly, but they can also perform complex management operations such as computing power, network traffic, disk usage limits, etc., and are no longer limited to simple multiplayer environments such as "I'm having fun playing online with my friends".
Again, an example.
A player clicks to buy another player's item on a server-hosted website, and an in-game drone activates and delivers the item he wants.
There is no doubt that this is very Cooooool, and I hope to see OC2 adopted on a large scale sooner rather than later.

Translated with www.DeepL.com/Translator (free version)

**Comment by samuelh2005** [comment]
Using Forge mods and Bukkit based plugins are not expected to work correctly and is not recommanded. The Bukkit API is not designed to work with modded content, so, you are very lucky if you don't get server crashes.

Please use Forge alternatives.
* Luckperms has a forge version
* There aren't many mods similar to PlotSquard but there is Open Parties and Claims

Read more: https://gist.github.com/Patbox/e44844294c358b614d347d369b0fc3bf


Also, OC2 robots do not use `FakePlayer`. Instead OC2 robots are their own Robot entity. This means permission plugins will ignore them.

**Reply by Mr-Ao-Dragon** [reply]
ok
